streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -353,19 +353,21 @@ public static StreamThread create(final TopologyMetadata topologyMetadata,
         final Map<String, Object> restoreConsumerConfigs = config.getRestoreConsumerConfigs(getRestoreConsumerClientId(threadId));
         final Consumer<byte[], byte[]> restoreConsumer = clientSupplier.getRestoreConsumer(restoreConsumerConfigs);
 
+        final boolean stateUpdaterEnabled =
+                InternalConfig.getBoolean(config.originals(), InternalConfig.STATE_UPDATER_ENABLED, false);
+
         final StoreChangelogReader changelogReader = new StoreChangelogReader(
             time,
             config,
```

Comment:
nit: we already pass the actual configs into the changelog reader, why not set the `stateUpdaterEnabled` flag inside the constructor/StoreChangelogReader class?